exp/controllers/azuremachinepool_annotations.go

```diff
@@ -24,7 +24,7 @@ import (
 // `content`. `content` in this case should be a `map[string]interface{}`
 // suitable for turning into JSON. This `content` map will be marshalled into a
 // JSON string before being set as the given `annotation`.
-func (r *AzureMachinePoolReconciler) updateAnnotationJSON(rw annotationReaderWriter, annotation string, content map[string]interface{}) error {
+func (r *AzureMachinePoolReconciler) updateAnnotationJSON(rw annotationReaderWriter, annotation string, content map[string]interface{}) error { // nolint:unused // This function is not used at the moment, if you use it, please delete the comment explaining
```

Comment:
Removing the func or commenting "nolint" will fix the unused ones.

I don't know if this func is still necessary, so I'll try nolint first, what do you think?

Reply:
I don't see any magical invocations anywhere. I'd vote for delete, and run the full e2e tests to verify behavior.

Reply:
> I don't see any magical invocations anywhere. I'd vote for delete, and run the full e2e tests to verify behavior.

Yes, it can be removed, but the e2e test I can't run locally, can I do it via CI?

Reply:
You can run e2e tests locally. Give the [docks on e2e tests](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/master/docs/development.md#e2e-testing) a perusal.